components/authz-service/server/v2/policy.go

```diff
@@ -859,8 +869,18 @@ func statementFromAPI(statement *api.Statement) (storage.Statement, error) {
 		statement.Resources = []string{"*"}
 	}
 
-	// map external representation of "all projects" to actual ID for that meta-project
 	projects := make([]string, len(statement.Projects))
+
+	if len(statement.Projects) == 0 {
+		if s.isBeta2p1() {
+			// no projects is an error for v2.1
+			return v2.Statement{}, errors.New("policy statements must include projects")
+		}
+		// no projects OK for v2.0 but set to "all" to allow 2.1 upgrade to work when needed
+		projects = append(projects, constants.AllProjectsID)
+	}
+
+	// map external representation of "all projects" to actual ID for that meta-project
 	for i, project := range statement.Projects {
```

Comment:
I wonder if we now should be iterating over `projects` and not `statement.Projects` . . . in which case, we'd want to append the `AllProjectsExternalID`